apps/ledger-live-mobile/src/screens/ReceiveFunds/01-SelectCrypto.tsx

```diff
@@ -0,0 +1,152 @@
+// @flow
+import React, { useCallback, useMemo } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { StyleSheet, FlatList } from "react-native";
+import type {
+  CryptoCurrency,
+  TokenCurrency,
+} from "@ledgerhq/live-common/lib/types";
+import {
+  isCurrencySupported,
+  listTokens,
+  useCurrenciesByMarketcap,
+  listSupportedCurrencies,
+} from "@ledgerhq/live-common/lib/currencies";
+
+import {Flex } from "@ledgerhq/native-ui";
+import { useSelector } from "react-redux";
+import { Account, TokenAccount } from "@ledgerhq/live-common/src/types";
+import { makeEmptyTokenAccount } from "@ledgerhq/live-common/src/account";
+import { useRoute } from "@react-navigation/native";
+import { ScreenName } from "../../const";
+import { track, TrackScreen } from "../../analytics";
+import FilteredSearchBar from "../../components/FilteredSearchBar";
+import CurrencyRow from "../../components/CurrencyRow";
+import LText from "../../components/LText";
+import { flattenAccountsSelector } from "../../reducers/accounts";
+import { usePreviousRouteName } from "../../helpers/routeHooks";
+
+const SEARCH_KEYS = ["name", "ticker"];
+
+type Props = {
+  devMode: boolean,
+  navigation: any,
+  route: { params: { filterCurrencyIds?: string[] } },
+};
+
+const keyExtractor = (currency: CryptoCurrency | TokenCurrency) => currency.id;
+
+const renderEmptyList = () => (
+  <Flex px={6}>
+    <LText textAlign="center">
+      <Trans i18nKey="common.noCryptoFound" />
+    </LText>
+  </Flex>
+);
+
+const listSupportedTokens = () =>
+  listTokens().filter(t => isCurrencySupported(t.parentCurrency));
+
+const findAccountByCurrency = (accounts: (TokenAccount | Account)[], currency: CryptoCurrency | TokenCurrency) => accounts.filter((acc: TokenAccount | Account) =>
+    (acc.type === "Account" ? acc.currency?.id : acc.token.id) === currency.id
+  )
+
+export default function AddAccountsSelectCrypto({ navigation, route }: Props) {
+  const {t } = useTranslation();
+  const routerRoute = useRoute()
+  const { filterCurrencyIds = [] } = route.params || {};
+  const lastRoute = usePreviousRouteName()
+  const cryptoCurrencies = useMemo(
+    () =>
+      listSupportedCurrencies()
+        .concat(listSupportedTokens())
+        .filter(
+          ({ id }) =>
+            filterCurrencyIds.length <= 0 || filterCurrencyIds.includes(id),
+        ),
+    [filterCurrencyIds],
+  );
+
+  const accounts = useSelector(flattenAccountsSelector);
+
+  const sortedCryptoCurrencies = useCurrenciesByMarketcap(cryptoCurrencies);
+
+  const onPressItem = useCallback((currency: CryptoCurrency | TokenCurrency) => {
+    track('currency_clicked', {screen: routerRoute.name, currency: currency.name})
+
+    const accs = findAccountByCurrency(accounts, currency);
+    if(accs.length > 1) {
```

Comment:
a lot of if else. I have feeling code here could be improved.
alternatively, please have code comment because it's very hard to read what's the code is doing.
especially there is a nesting of if else on 2 levels.